testing/regress/ecl/trim.ecl

```diff
@@ -0,0 +1,60 @@
+/*##############################################################################
+
+    HPCC SYSTEMS software Copyright (C) 2018 HPCC Systems®.
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+       http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+// Test variations: left, right, both, all
+// Test variations: normal, whitespace
+// Test variations: utf8, string, varstring, unicode, varunicode
+// Test variations: const folded, not const folded
+// Test variations: spaces present, spaces absent, whitespace present, empty, all whitespace
+
+declare_tests(testtype) := MACRO
+#uniquename(input)
+#uniquename(output)
+#uniquename(trans)
+%input% := NOFOLD(DATASET([{''},{' '},{' \t a b\tc \t '},{'abc'}], { testtype s; }));
```

Comment:
It would be good to test folded and non folded examples.  E.g.
 
    %rawinput% := DATASET([{''},{' '},{' \t a b\tc \t '},{'abc'}], { testtype s; });
    %input% := NOFOLD(%rawinput%);

    OUTPUT(project(%rawinput%, %trans%(LEFT)));
    OUTPUT(project(%input%, %trans%(LEFT)));

I tried, and I think it revealed an issue.